### PR TITLE
Create a two-shim jar by default on Databricks [databricks]

### LINKED ISSUE #9008: [FEA] Make databricks CI pipeline use a multi-shim jar.
**Is your feature request related to a problem? Please describe.**
As a Spark RAPIDS developer, I need the costly CI pipeline to catch Shim-related issues. Databricks environment requires most complicated shimming. but uses a single-shim jar.  Issues like the build failure #8962 and some runtime issues are caught only with a multi-shim build.

**Describe the solution you'd like**
Bulld at least a two-shim jar 311 + 3XYdb to be used in the Databricks test pipelines. 

**Describe alternatives you've considered**
- Manual testing 
- Delayed test coverage via nightly
- Release dedicated conventional jar artifacts for cloud environments

**Additional context**
#8962

### PR DESCRIPTION
**Pull request by gerashegalov**
Fixes #9008

Testing:

- [x] reproduces #8962 
- [x] build.sh and test.sh on top of #9006 works

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

File: jenkins/databricks/build.sh [hunk 1]
```diff
@@ -143,5 +147,10 @@ fi
 # Build the RAPIDS plugin by running package command for databricks
 mvn -B -Ddatabricks -Dbuildver=$BUILDVER clean package -DskipTests $MVN_OPT
 
+if [[ "$WITH_DEFAULT_UPSTREAM_SHIM" != "0" ]]; then
+    echo "Building the default Spark shim and creatign a two-shim dist jar"
```
Comment (tgravescs) [inline jenkins/databricks/build.sh:151]:
```suggestion
    echo "Building the default Spark shim and creating a two-shim dist jar"
```

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Review by jlowe — DISMISSED** [review]
Changes look OK to me, but wondering if we want to target 23.08 for this given how close we are to release.

**Review by razajafri — COMMENTED** [review]
LGTM

but I have the same comment as @jlowe why are we targeting 23.08?

**Comment by gerashegalov** [comment]
I share the concern re: 23.08. I am torn only because I am afraid that we might still need to make changes to 23.08 that will not receive this testing only in nightly. FWIW I already see the change is successfully exercised by the CI.

**Comment by gerashegalov** [comment]
build

**Review by razajafri — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Pull request merged**